dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventCategoryOptValidationHook.java

```diff
@@ -0,0 +1,126 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ObjectUtils;
+import org.hisp.dhis.category.CategoryOption;
+import org.hisp.dhis.category.CategoryOptionCombo;
+import org.hisp.dhis.i18n.I18nFormat;
+import org.hisp.dhis.i18n.I18nManager;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.tracker.TrackerImportStrategy;
+import org.hisp.dhis.tracker.domain.Event;
+import org.hisp.dhis.tracker.report.TrackerErrorCode;
+import org.hisp.dhis.tracker.report.ValidationErrorReporter;
+import org.hisp.dhis.tracker.validation.TrackerImportValidationContext;
+import org.hisp.dhis.util.DateUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.Objects;
+
+import static org.hisp.dhis.tracker.report.ValidationErrorReporter.newReport;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@Component
+@Slf4j
+public class EventCategoryOptValidationHook
+    extends AbstractTrackerDtoValidationHook
+{
+    @Autowired
+    protected I18nManager i18nManager;
+
+    public EventCategoryOptValidationHook()
+    {
+        super( Event.class, TrackerImportStrategy.CREATE_AND_UPDATE );
+    }
+
+    @Override
+    public void validateEvent( ValidationErrorReporter reporter, Event event )
+    {
+        TrackerImportValidationContext context = reporter.getValidationContext();
+
+        Program program = context.getProgram( event.getProgram() );
+        Objects.requireNonNull( program, Constants.PROGRAM_CANT_BE_NULL );
+        Objects.requireNonNull( context.getBundle().getUser(), Constants.USER_CANT_BE_NULL );
+        Objects.requireNonNull( program, Constants.PROGRAM_CANT_BE_NULL );
+        Objects.requireNonNull( event, Constants.EVENT_CANT_BE_NULL );
+
+        CategoryOptionCombo categoryOptionCombo = context
+            .getCategoryOptionCombo(
+                reporter.getValidationContext().getCachedEventCategoryOptionCombo( event.getUid() ) );
+
+        Objects.requireNonNull( categoryOptionCombo, Constants.CATEGORY_OPTION_COMBO_CANT_BE_NULL );
+
+        if ( categoryOptionCombo.isDefault()
+            && program.getCategoryCombo() != null
+            && !program.getCategoryCombo().isDefault() )
+        {
+            reporter.addError( newReport( TrackerErrorCode.E1055 ) );
+            return;
+        }
+
+        Date eventDate;
+        try
+        {
+            eventDate = DateUtils.parseDate( ObjectUtils
+                .firstNonNull( event.getOccurredAt(), event.getScheduledAt(), DateUtils.getIso8601( new Date() ) ) );
+        }
+        catch ( IllegalArgumentException e )
+        {
+            log.debug( "Failed to parse dates, an error should already be reported." );
```

Comment:
Is this handled somewhere? or its missing the error report?

Reply:
Yes, this will get into the error report at an earlier stage. So this is just to not fall out of the validation if something is broken.